logstash-core/lib/logstash/inputs/metrics.rb

```diff
@@ -39,7 +44,10 @@ def update(time, snapshot)
       #   - We can use a synchronization mechanism between the called thread (update method)
       #   and the plugin thread (run method)
       #   - How we handle back pressure here?
-      @queue << snapshot.to_event
+      #   - one snashot should be only one event
+      snapshot.to_events.each do |event|
```

Comment:
Agree this could be a costly operation if we have a large quantity of metrics, I think we need to benchmark this to find a best approach ?